address-sanitizer/tools/kasan_symbolize.py

```diff
@@ -137,13 +141,28 @@ def ProcessLine(self, line, lines_before, lines_after, questionable):
     symbolizer = self.module_symbolizers[module]
     loader = self.module_offset_loaders[module]
 
-    symbol_offset = loader.LookupOffset(function)
-    if not symbol_offset:
+    symbol_offsets = loader.LookupOffsets(function)
+    if len(symbol_offsets) == 0:
       print line
       return
+    elif len(symbol_offsets) == 1:
+      symbol_addr = symbol_offsets[0]
+    else:
+      # If there are more than one symbol with the same name, best-effort guess
+      # the correct one by comparing 12 (log(min(PAGE_SIZE))) least significant
+      # bits of the address. We can't use the whole address for this, since it
+      # is subject to randomization.
+      symbol_addr = None
+      for symbol_offset in symbol_offsets:
+        if (int(addr, 16) % (1 << 12)) == (symbol_offset % (1 << 12)):
```

Comment:
Unless I misread the code, symbol_offset is a beginning of function while addr points somewhere inside of the function, so this condition will never be true.
You need to use something like addr-off.
Also it would be nice to make it work reliably at least without ASLR. We need to check exact match first, otherwise guess. Also we could compare function size (present both in nm output and report, however note that nm and kernel usually slightly disagree on function size: nm uses symbol size, while kernel seems to do next_function_addr-current_function_addr).